### ISSUE TITLE
Backend Snapshots Filling up Disk

### ISSUE THREAD

**Issue by knoppiks** [ubuntu/microk8s#2167]
I'm running microk8s HA on 3 Raspberry PI 4 using Ubuntu 20.04.

I recently noticed that one of my nodes evicted all pods due to **ephemeral storage** shortage. Inspecting the node showed high disk usage of `/var/snap/microk8s/2097/var/kubernetes/backend/`. A lot of "snapshot" files filling up over 20GB of my 32GB System SD Card. 

Is there a GC mechanism to remove old backend snapshots?
Should I add one myself?

**Comment by ktsakalozos** [comment]
Hi @knoppiks,  this issue is already reported in https://github.com/canonical/dqlite/issues/289 and it has been fixed https://github.com/canonical/raft/pull/193 a few days ago. MicroK8s from track v1.21 has already the fix, the rest of the track will get the fix on the next patch release (eg 1.20.6 for the 1.20 track).

While waiting for the fix to be released you can perform a manual clean-up. In  `/var/snap/microk8s/current/var/kubernetes/backend/` you will see snapshot files that normally have a corresponding `.meta file, eg:
```
-rw-rw---- 1 root microk8s 6136784 Apr 12 11:28 snapshot-1-5120-642945
-rw-rw---- 1 root microk8s      72 Apr 12 11:28 snapshot-1-5120-642945.meta
-rw-rw---- 1 root microk8s 7356304 Apr 12 11:33 snapshot-1-6144-938765
-rw-rw---- 1 root microk8s      72 Apr 12 11:33 snapshot-1-6144-938765.meta
```

If you find a snapshot file *without* a matching `.meta` you can remove it.

Apologies for the inconvenience.

**Comment by knoppiks** [comment]
Thanks a lot!
